x-pack/test/functional/services/cases/common.ts

```diff
@@ -0,0 +1,152 @@
+  const header = getPageObject('header');
+  return {
+    /**
+     * Opens the create case page pressing the "create case" button.
+     *
+     * Doesn't do navigation. Only works if you are already inside a cases app page.
+     * Does not work with the cases flyout.
+     */
+    async openCreateCasePage() {
+      await testSubjects.click('createNewCaseBtn');
+      await testSubjects.existOrFail('create-case-submit', {
+        timeout: 5000,
+      });
+    },
+
+    /**
+     * it creates a new case from the create case page
+     * and leaves the navigation in the case view page
+     *
+     * Doesn't do navigation. Only works if you are already inside a cases app page.
+     * Does not work with the cases flyout.
+     */
+    async createCaseFromCreateCasePage(caseTitle: string = 'test-' + uuid.v4()) {
+      await this.openCreateCasePage();
+
+      // case name
+      await testSubjects.setValue('input', caseTitle);
+
+      // case tag
+      await comboBox.setCustom('comboBoxInput', 'tagme');
+
+      // case description
+      const descriptionArea = await find.byCssSelector('textarea.euiMarkdownEditorTextArea');
+      await descriptionArea.focus();
+      await descriptionArea.type('Test description');
+
+      // save
+      await testSubjects.click('create-case-submit');
+
+      await testSubjects.existOrFail('case-view-title');
+    },
+
+    /**
+     * Goes to the first case listed on the table.
+     *
+     * This will fail if the table doesn't have any case
+     */
+    async goToFirstListedCase() {
+      await testSubjects.existOrFail('cases-table');
+      await testSubjects.click('case-details-link');
+      await testSubjects.existOrFail('case-view-title');
+    },
+
+    /**
+     * Marks a case in progress via the status dropdown
+     */
+    async markCaseInProgressViaDropdown() {
+      await this.openCaseSetStatusDropdown();
+
+      await testSubjects.click('case-view-status-dropdown-in-progress');
+
+      // wait for backend response
+      await testSubjects.existOrFail('header-page-supplements > status-badge-in-progress', {
+        timeout: 5000,
+      });
+    },
+
+    /**
+     * Marks a case closed via the status dropdown
+     */
+    async markCaseClosedViaDropdown() {
+      this.openCaseSetStatusDropdown();
+
+      await testSubjects.click('case-view-status-dropdown-closed');
+
+      // wait for backend response
+      await testSubjects.existOrFail('header-page-supplements > status-badge-closed', {
+        timeout: 5000,
+      });
+    },
+
+    /**
+     * Marks a case open via the status dropdown
+     */
+    async markCaseOpenViaDropdown() {
+      this.openCaseSetStatusDropdown();
+
+      await testSubjects.click('case-view-status-dropdown-open');
+
+      // wait for backend response
+      await testSubjects.existOrFail('header-page-supplements > status-badge-open', {
+        timeout: 5000,
+      });
+    },
+
+    async deleteAllBulkListAction() {
```

Comment:
I do not understand what this function does from the naming. Maybe is better to say `bulkDeleteAllCases`